src/lib/API.js

```diff
@@ -0,0 +1,59 @@
+export function getCharacter() {
+  let url;
+  const filter = document.querySelector(".header__filter");
+  const status = filter.value;
+  const inputName = document.querySelector(".header__search-form");
+  const searchName = inputName.value;
+
+  if (status == "Alive") {
+    url = `https://rickandmortyapi.com/api/character?name=${searchName}&status=alive`;
+  } else if (status == "Dead") {
+    url = `https://rickandmortyapi.com/api/character?name=${searchName}&status=dead`;
+  } else if (status == "Unknown") {
+    url = `https://rickandmortyapi.com/api/character?name=${searchName}&status=unknown`;
+  } else if (status == "All") {
+    url = `https://rickandmortyapi.com/api/character?name=${searchName}`;
+  }
+
+  console.log(url);
+
+  fetch(url)
+    .then((response) => {
+      if (response.ok) {
+        return response.json();
+      } else {
+        console.log(response.status);
+      }
+    })
+    .then((dataObject) =>
+      dataObject.results.forEach((character) => {
+        const main = document.querySelector("main");
+        const section = document.createElement("section");
+        section.classList.add("character");
+        const fullName = document.createElement("h2");
+        fullName.classList.add("fullName");
+        fullName.textContent = `${character.name}`;
+        const img = document.createElement("img");
+        img.classList.add("img");
+        img.src = character.image;
+        img.alt = `$(character.name)`;
+
```

Comment:
Here is an advice i got as well:
to shorten your higher order function you could separate lines 29 to 53 and declare several functions for creating the html-elements. Then just call them in your second "then" and append the elements.

Reply:
and as you mentioned, you already plan to do that. Sorry have read your comment just after review.